### PR TITLE
Fixes the memory in JSONSchemaVisitor

### PR DESCRIPTION
**Pull request by w-vi**

### CODE REVIEW

File: src/refract/JSONSchemaVisitor.cc [hunk 1]
```diff
@@ -269,18 +282,18 @@ namespace refract
                         // want them in the schema, otherwise skip
                         // empty ones
                         if (allEmpty) {
-                            JSONSchemaVisitor renderer;
+                            JSONSchemaVisitor v;
```
Comment (pksunkara) [inline src/refract/JSONSchemaVisitor.cc]: Please keep the original name `renderer`.

File: src/refract/JSONSchemaVisitor.cc [hunk 2]
```diff
@@ -166,15 +174,17 @@ namespace refract
...
 
             if (elm->element() == "enum"){
-                v.addMember("enum", elm);
+                IElement *e = NULL;
+                e = elm->clone();
+                v.addMember("enum", e);
             }
             else if (elm->element() != "object") {
+                ArrayElement *enm = new ArrayElement;
                 for(std::vector<IElement*>::const_iterator i = items.begin();
```
Comment (pksunkara) [inline src/refract/JSONSchemaVisitor.cc]: I missed this previously. `for (`

File: src/refract/JSONSchemaVisitor.cc [hunk 3]
```diff
@@ -191,9 +201,12 @@ namespace refract
                 if (!enm->empty()) {
                     v.addMember("enum", enm);
                 }
+                else{
```
Comment (pksunkara) [inline src/refract/JSONSchemaVisitor.cc]: `else {`

File: src/refract/JSONSchemaVisitor.h [hunk 4]
```diff
@@ -50,6 +51,7 @@ namespace refract
         void visit(const BooleanElement& e);
 
         IElement* get();
```
Comment (pksunkara) [inline src/refract/JSONSchemaVisitor.h]: Is this function not used anymore?

File: src/refract/JSONSchemaVisitor.cc [hunk 5]
```diff
@@ -134,9 +141,7 @@ namespace refract
                 renderer.visit(*(*it));
 
                 ObjectElement *o1 = TypeQueryVisitor::as<ObjectElement>(renderer.get());
-                MemberElement *m1 = TypeQueryVisitor::as<MemberElement>(o1->value[0]);
-
-                o1->renderType(IElement::rCompact);
+                MemberElement *m1 = TypeQueryVisitor::as<MemberElement>(o1->value[0]->clone());
```
Comment (pksunkara) [inline src/refract/JSONSchemaVisitor.cc]: Please add an empty line after this.

**Comment by pksunkara** [comment]
Just minor stuff.

File: src/refract/JSONSchemaVisitor.cc (hunk 1 above)
Comment (w-vi) [inline src/refract/JSONSchemaVisitor.cc]: Variable shadowing takes placd hence the renaming and becase the scope is 3 lines i wantdd something short

File: src/refract/JSONSchemaVisitor.h (hunk 4 above)
Comment (w-vi) [inline src/refract/JSONSchemaVisitor.h]: It actually stil is used, otherwise I would remove it

**Pull request merged**